### PR TITLE
Properly validate dependent type annotations on class declarations.

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java [hunk 1]
```diff
@@ -254,6 +255,34 @@ public void standardizeReturnType(MethodTree m, AnnotatedTypeMirror atm) {
...
+     * @param node the class declaration
+     * @param type the type of the class declaration
+     * @param ele the element of the class declaration
+     */
+    public void standardizeClass(ClassTree node, AnnotatedTypeMirror type, Element ele) {
+        if (!hasDependentType(type)) {
+            return;
+        }
+        TreePath path = factory.getPath(node);
+        if (path == null) {
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: When would the path for a `ClassTree` be null?

Comment (wmdietl) [reply framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: I don't know why this code checked for null originally.
We could separately try cleaning these null checks up.

Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: OK, feel free to go ahead.

**Comment by mernst** [comment]
Tests are failing because checker-framework-inference needs to be adapted to the new names.

File: framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java (hunk 1 above)
Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java]: Or, just leave it as is; someone else will have to do the cleanup in the future.

**Pull request merged**